tracer/src/Datadog.Trace.Tools.dd_dotnet/CreatedumpCommand.cs

```diff
@@ -0,0 +1,676 @@
+            {
+                crashThread = (int)firstThreadWithException.OSThreadId;
+            }
+        }
+
+        ClrException? exception = null;
+
+        // Check if there's an exception on the crash thread
+        if (crashThread != null)
+        {
+            exception = _runtime.Threads.FirstOrDefault(t => t.OSThreadId == crashThread.Value)?.CurrentException;
+        }
+
+        bool isSuspicious;
+
+        try
+        {
+            var callback = (delegate* unmanaged<IntPtr, ResolveMethodData*, int>)&ResolveManagedMethod;
+            crashReport.ResolveStacks(crashThread ?? 0, (IntPtr)callback, out isSuspicious);
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to resolve stacks: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        if (!isSuspicious && exception != null)
+        {
+            // The stacks aren't suspicious, but maybe the exception is
+            var exceptionType = exception.Type?.Name ?? string.Empty;
+
+            var suspiciousExceptionTypes = new[] { "System.InvalidProgramException", "System.Security.VerificationException", "System.MissingMethodException", "System.BadImageFormatException" };
+
+            if (exceptionType.StartsWith("Datadog") || suspiciousExceptionTypes.Contains(exceptionType))
+            {
+                isSuspicious = true;
+            }
+        }
+
+        if (!isSuspicious)
+        {
+            return;
+        }
+
+        AnsiConsole.WriteLine("Datadog - The crash might have been caused by automatic instrumentation, sending crash report...");
+
+        if (signal.HasValue)
+        {
+            _ = SetSignal(crashReport, signal.Value);
+        }
+
+        _ = SetMetadata(crashReport, _runtime, exception);
+
+        try
+        {
+            var outputFile = Environment.GetEnvironmentVariable("DD_TRACE_CRASH_OUTPUT");
+
+            if (!string.IsNullOrEmpty(outputFile))
+            {
+                var path = IntPtr.Zero;
+
+                try
+                {
+                    AnsiConsole.WriteLine($"Writing crash report to {outputFile}...");
+                    path = Marshal.StringToHGlobalAnsi(outputFile);
+                    crashReport.WriteToFile(path);
+                }
+                finally
+                {
+                    if (path != IntPtr.Zero)
+                    {
+                        Marshal.FreeHGlobal(path);
+                    }
+                }
+            }
+            else
+            {
+                crashReport.Send();
+            }
+        }
+        catch (Win32Exception ex)
+        {
+            Errors.Add($"Failed to send crash report: {GetLastError(crashReport, ex)}");
+            return;
+        }
+
+        AnsiConsole.WriteLine("Datadog - Crash report sent successfully");
+    }
+
+    private void InvokeCreatedump(string createdumpPath)
+    {
+        if (File.Exists(createdumpPath))
```

Comment:
Tony had to do some stuff to try to find the process (using `which`?), even if `!File.Exists()` to replicate the default behaviour you get without interception. Maybe we need to do the same here?

Reply:
.NET is supposed to give the absolute path to createdump, so no trickery needed